### PR TITLE
Illustrates flash-message concept in `insert`

### PR DESCRIPTION
**Pull request by manandearth**
## Goal
To implement a flash-message on `insert`, `delete`,  and `update` with the 302 to `/invoices`

### CODE REVIEW

**Comment by manandearth** [comment]
There is a pedestal flash interceptor that I try to call 
`io.pedestal.http.ring-middlewares/flash `
It wraps the `ring.middleware.flash` namespace.
>   "If a :flash key is set on the response by the handler, a :flash key with                                                                        
  the same value will be set on the next request that shares the same session.                                                                     
  This is useful for small messages that persist across redirects."       

I don't know how a `:flash` added in `insert` endpoint can be carried to `retrieve-all` view, since the 302 redirect doesn't. (am I right?)

**Comment by vemv** [comment]
> io.pedestal.http.ring-middlewares/flash 

Good find!

> I don't know how a :flash added in insert endpoint can be carried to retrieve-all view

Flashes work by setting a cookie. Cookies persist across requests.

**Comment by manandearth** [comment]
so I guess: [my lesson in buddy, authentication and cookies](https://lambdaisland.com/episodes/buddy-authentication#/transcript) is due...

**Comment by vemv** [comment]
Sounds good. Although 'cookies' are a subset of 'authentication' in terms of what you need to learn.

i.e. you can go for something simpler at first

I recommend understanding this doc well https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies

**Comment by manandearth** [comment]
Simple first stage (I think)

File: src/playground/service.clj [hunk 1]
```diff
@@ -7,6 +7,7 @@
    [io.pedestal.http.body-params :as body-params]
    [io.pedestal.http.route :as route]
    [io.pedestal.interceptor.chain :as interceptor-chain]
+   [io.pedestal.http.ring-middlewares :as middlewares]
```
Comment (vemv) [inline src/playground/service.clj]: Please follow these: https://stuartsierra.com/2015/05/10/clojure-namespace-aliases

In this case, I'd use `ring-middlewares` or `pedestal.http.ring-middlewares`, or no alias at all.

File: src/playground/services/invoices/retrieve_all/view.clj [hunk 2]
```diff
@@ -5,11 +5,12 @@
    [io.pedestal.http.route :refer [url-for]]
    [playground.views :as views]))
 
-(defn all-invoices [context]
+(defn all-invoices [context & info]
   (page/html5
    (views/gen-page-head "All Entries")
    views/header-links
    [:div
+    [:h2 info]
```
Comment (vemv) [inline src/playground/services/invoices/retrieve_all/view.clj]: `(when (seq info) [:h2 info])`

**Review by vemv — COMMENTED** [review]
Looking nice!

* You should be using the  `flash` middleware. As it name suggests, its effect is transient: after refreshing the page, the message should be gone.

* Make sure to use commit messages in imperative and capitalized as in https://github.com/zenmacs/.emacs.d/commits/master
  * I had said 'present simple' in the past, my fault :)

File: src/playground/service.clj (hunk 1 above)
Comment (manandearth) [reply src/playground/service.clj]: I understand and agree. The `middlewares` alias I saw in pedestal samples, where I found the cookie pattern. https://github.com/pedestal/pedestal/blob/a99159ef221c4fb1055d74fece93088890d7d32e/samples/ring-middleware/src/ring_middleware/service.clj#L18

**Comment by manandearth** [comment]
The messages work now. Perhaps more can be done here? or more in a cookie....

**Review by vemv — APPROVED** [review]

**Pull request merged**